Buildfarm worker not starting when following quickstart guide

Platform: Ubuntu 

When following the [quick start guide](https://bazelbuild.github.io/bazel-buildfarm/docs/quick_start/), attempting to test remote caching, there is an error when using the `bazel run --remote_cache=grpc://localhost:8980 :main` command. 

I am running the buildfarm server using `bazelisk run src/main/java/build/buildfarm:buildfarm-server $PWD/examples/config.minimal.yml`
According to the guide this should allow me to cache. 

Server Error : 
`Sep 21, 2023 11:38:30 AM build.buildfarm.common.services.ByteStreamService queryWriteStatus
SEVERE: queryWriteStatus(uploads/257bd951-35ea-49ad-8b50-79579f8546c2/blobs/3fed0759c1fc6673599af35c1cd4272c04563cbae7f98e969deaed0b70e0c4b5/165)
com.google.common.util.concurrent.UncheckedExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: no available workers
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2086)
        at com.google.common.cache.LocalCache.get(LocalCache.java:4012)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4035)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5011)
        at build.buildfarm.instance.shard.Writes.get(Writes.java:151)
        at build.buildfarm.instance.shard.ShardInstance.getBlobWrite(ShardInstance.java:1246)
        at build.buildfarm.common.services.ByteStreamService.getUploadBlobWrite(ByteStreamService.java:434)
        at build.buildfarm.common.services.ByteStreamService.getWrite(ByteStreamService.java:447)
        at build.buildfarm.common.services.ByteStreamService.queryWriteStatus(ByteStreamService.java:361)
        at com.google.bytestream.ByteStreamGrpc$MethodHandlers.invoke(ByteStreamGrpc.java:316)
        at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
        at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
        at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
        at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
        at io.grpc.util.TransmitStatusRuntimeExceptionInterceptor$1.onHalfClose(TransmitStatusRuntimeExceptionInterceptor.java:74)
        at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
        at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
        at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
        at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:346)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:860)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: no available workers
        at io.grpc.Status.asRuntimeException(Status.java:530)
        at build.buildfarm.instance.shard.ShardInstance.getRandomWorker(ShardInstance.java:1197)
        at build.buildfarm.instance.shard.ShardInstance.writeInstanceSupplier(ShardInstance.java:1185)
        at build.buildfarm.instance.shard.Writes$1.load(Writes.java:132)
        at build.buildfarm.instance.shard.Writes$1.load(Writes.java:128)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3571)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2313)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2190)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2080)
        ... 25 more`

Project Error : 
`ERROR: /home/harv/buildfarm/project/BUILD.bazel:1:10: Compiling main.cc failed: (Exit 34): 7 errors during bulk transfer:
java.io.IOException: Error while uploading artifact with digest 'ee49cf3e151739f8c6d96ceeace06dfb58b768bbaba97a7d1b1c38cfdc80c858/89'
java.io.IOException: Error while uploading artifact with digest 'c0f9c8978ac2021eecf5901efb80e709688674be031c98893bf9bcceeda852d0/566'
java.io.IOException: Error while uploading artifact with digest 'a91420b3068b4230b9de411b2b97ec86db6b56550c4b9cc605be3e0b533a0cca/144'
java.io.IOException: Error while uploading artifact with digest '68e8c623ae229eb15e35ce74a57223ba2d468e51738fa7bf46ef8648af9b36cf/1854'
java.io.IOException: Error while uploading artifact with digest '63cc83ed5ea84ff501397d90af8223168270dda5fe403bb775e79551d1175b64/108'
java.io.IOException: Error while uploading artifact with digest '0bd5b520c26cc0369772e0bb3914a99446661f1d52466d7c2c0643b244181f5b/104'
java.io.IOException: Error while uploading artifact with digest '3fed0759c1fc6673599af35c1cd4272c04563cbae7f98e969deaed0b70e0c4b5/165'
Target //:main failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 4.762s, Critical Path: 3.48s
INFO: 5 processes: 5 internal.
FAILED: Build did NOT complete successfully`

--- a/_site/docs/quick_start.md
+++ b/_site/docs/quick_start.md
@@ -10,7 +10,7 @@ Here we describe how to use bazel remote caching or remote execution with buildf
 
 ## Setup
 
-You can run this quick start on a single computer running nearly any flavor of linux. This computer is the localhost for the rest of the description.
+You can run this quick start on a single computer running any flavor of linux that bazel supports. A C++ compiler is used here to demonstrate action execution. This computer is the localhost for the rest of the description.
 
 ### Backplane
 
@@ -44,7 +44,7 @@ cc_binary(
 
 And an empty WORKSPACE file.
 
-As a test, verify that `bazel run :main` builds your main program and runs it, and prints `Hello, World!`. This will ensure that you have properly installed bazel and a C++ compiler, and have a working target before moving on to remote execution.
+As a test, verify that `bazel run :main` builds your main program and runs it, and prints `Hello, World!`. This will ensure that you have properly installed `bazel` and a C++ compiler, and have a working target before moving on to remote caching or remote execution.
 
 Download and extract the buildfarm repository. Each command sequence below will have the intended working directory indicated, between the client (workspace running bazel), and buildfarm.
 
@@ -52,25 +52,33 @@ This tutorial assumes that you have a bazel binary in your path and you are in t
 
 ## Remote Caching
 
-A Buildfarm server with an instance can be used strictly as an ActionCache and ContentAddressableStorage to improve build performance. This is an example of running a bazel client that will retrieve results if available, and store them if the cache is missed and the execution needs to run locally.
+A Buildfarm cluster can be used strictly as an ActionCache (AC) and ContentAddressableStorage (CAS) to improve build performance. This is an example of running a bazel client that will retrieve results if available, otherwise store them on a cache miss after executing locally.
 
 Download the buildfarm repository and change into its directory, then:
 
-run `bazelisk run src/main/java/build/buildfarm:buildfarm-server $PWD/examples/config.minimal.yml`
+ * run `bazel run src/main/java/build/buildfarm:buildfarm-server $PWD/examples/config.minimal.yml`
 
 This will wait while the server runs, indicating that it is ready for requests.
 
-From another prompt (i.e. a separate terminal) in your newly created workspace directory from above:
+A server alone does not itself store the content of action results. It acts as an endpoint for any number of workers that present storage, so we must also start a single worker.
 
-run `bazel clean`
-run `bazel run --remote_cache=grpc://localhost:8980 :main`
+From another prompt (i.e. a separate terminal) in the buildfarm repository directory:
+
+ * run `bazel run src/main/java/build/buildfarm:buildfarm-shard-worker $PWD/examples/config.minimal.yml`
+
+This will also wait while the worker runs, indicating it will be available to store cache content.
+
+From another prompt in your newly created workspace directory from above:
+
+ * run `bazel clean`
+ * run `bazel run --remote_cache=grpc://localhost:8980 :main`
 
 Why do we clean here? Since we're verifying re-execution and caching, this ensures that we will execute any actions in the `run` step and interact with the remote cache. We should be attempting to retrieve cached results, and then when we miss - since we just started this memory resident server - bazel will upload the results of the execution for later use. There will be no change in the output of this bazel run if everything worked, since bazel does not provide output each time it uploads results.
 
 To prove that we have placed something in the action cache, we need to do the following:
 
-run `bazel clean`
-run `bazel run --remote_cache=localhost:8980 :main`
+ * run `bazel clean`
+ * run `bazel run --remote_cache=localhost:8980 :main`
 
 This should now print statistics on the `processes` line that indicate that you've retrieved results from the cache for your actions:
 
@@ -80,20 +88,22 @@ INFO: 2 processes: 2 remote cache hit.
 
 ## Remote Execution (and caching)
 
-Now we will use buildfarm for remote execution with a minimal configuration - a single memory instance, with a worker on the localhost that can execute a single process at a time - via a bazel invocation on our workspace.
+Now we will use buildfarm for remote execution with a minimal configuration with a worker on the localhost that can execute a single process at a time, via a bazel invocation on our workspace.
 
-First, we should restart the buildfarm server to ensure that we get remote execution (this can also be forced from the client by using `--noremote_accept_cached`). From the buildfarm server prompt and directory:
+First, we should restart the buildfarm server, and delete the worker's cas storage to ensure that we get remote execution (this can also be forced from the client by using `--noremote_accept_cached`). From the buildfarm server prompt and directory:
 
-interrupt a running `buildfarm-server`
-run `bazelisk run src/main/java/build/buildfarm:buildfarm-server $PWD/examples/config.minimal.yml`
+ * interrupt the running `buildfarm-server` (i.e. Ctrl-C)
+ * run `bazel run src/main/java/build/buildfarm:buildfarm-server $PWD/examples/config.minimal.yml`
 
 From another prompt in the buildfarm repository directory:
 
-run `bazelisk run src/main/java/build/buildfarm:buildfarm-shard-worker $PWD/examples/config.minimal.yml`
+ * interrupt the running `buildfarm-shard-worker` (i.e. Ctrl-C)
+ * recursively remove the worker root directory: from the config.minimal.yml, this will be `rm -fr /tmp/worker`
+ * run `bazel run src/main/java/build/buildfarm:buildfarm-shard-worker $PWD/examples/config.minimal.yml`
 
 From another prompt, in your client workspace:
 
-run `bazel run --remote_executor=grpc://localhost:8980 :main`
+ * run `bazel run --remote_executor=grpc://localhost:8980 :main`
 
 Your build should now print out the following on its `processes` summary line:
 
@@ -116,6 +126,10 @@ To stop the containers, run:
 ```
 ./examples/bf-run stop
 ```
+
+## Next Steps
+
+We've started our worker on the same host as our server, and also the same host on which we built with bazel, but these services can be spread across many machines, per 'remote'. A large number of workers, with a relatively small number of servers (10:1 and 100:1 ratios have been used in practice), consolidating large disks and beefy multicore cpus/gpus on workers, with specialization of what work they perform for bazel builds (or other client work), and specializing servers to have hefty network connections to funnel content traffic. A buildfarm deployment can service hundreds or thousands of developers or CI processes, enabling them to benefit from each others' shared context in the AC/CAS, and the pooled execution of a fleet of worker hosts eager to consume operations and deliver results.
 
 ## Buildfarm Manager
 

